### PR TITLE
Refactor lcm

### LINKED ISSUE #809: Refactor LCM the same way as GCD

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
Fixes #809. Don't ask about the commit history for this one, squashing will get rid of all sins :).

### CODE REVIEW

**Comment by JacquesCarette** [comment]
Very nice!

But apparently this also informs me that I don't understand what `abstract` really does. I've used it, as you document, to prevent unhelpful unfoldings that make proofs harder than necessary. But doesn't this do-not-unfold translate into do-not-compute too? O(1) is indeed faster than anything else... So perhaps what I don't understand is how `abstract` interacts with compilation.

So I guess I should also ask: how do I get `lcm` (say) to compute in the interpreter?

**Comment by MatthewDaggitt** [comment]
> But doesn't this do-not-unfold translate into do-not-compute too? O(1) is indeed faster than anything else... So perhaps what I don't understand is how abstract interacts with compilation.
>
> So I guess I should also ask: how do I get lcm (say) to compute in the interpreter?

Hmm so I'd always assumed that `abstract` only blocked reduction during type checking and normalisation in the editor, but when compiled the code ran as is. Having looked through the [documentation](https://agda.readthedocs.io/en/v2.6.0.1/language/abstract-definitions.html) though, I can't see anything written that supports or disproves that assumption.

Actually you make an excellent point that even if it does compute, we perhaps don't want to mark it abstract as many people like to normalise terms in the editor. In particular this will prevent rational calculations. I'll revert it.

**Comment by JacquesCarette** [comment]
You might want to ascertain the actual behaviour of `abstract` before you do.

The other way to get similar behaviour is to go through module parameters.

File: CHANGELOG.md [hunk 1]
```diff
@@ -605,10 +604,11 @@ Other minor additions
   [1+a%n]≤1+m⇒[a%n]≤m : 0 < suc n % d → suc n % d ≤ suc m → n % d ≤ m
 
   0/n≡0           : 0 / n ≡ 0
-  n/1≡a           : n / 1 ≡ n
+  n/1≡n           : n / 1 ≡ n
```
Comment (rei1024) [inline CHANGELOG.md]: The name of this proof also needs to be changed.
https://github.com/agda/agda-stdlib/blob/ef31b4238066c921f3d4b66feffdf97202aebd02/src/Data/Nat/DivMod.agda#L133-L134

Comment (MatthewDaggitt) [reply CHANGELOG.md]: Thanks, changed!

**Comment by MatthewDaggitt** [comment]
Removed abstract keywords to allow normalisation of rational arithmetic.

**Pull request merged**